src/Reprocessing_Matrix.re

```diff
@@ -74,4 +77,63 @@ let matvecmul = (m, v) => {
  [3 4 5] * [y] = [x3 + y4 + 5]
  [6 7 8]   [1]   [ who cares ]
  */
-let matptmul = (m, (x, y)) => (x *. m[0] +. y *. m[1] +. m[2], x *. m[3] +. y *. m[4] +. m[5]);
+let matptmul = (m, (x, y)) => (
+  x *. m[0] +. y *. m[1] +. m[2],
+  x *. m[3] +. y *. m[4] +. m[5]
+);
+
+
+/***
+ Invert a matrix
+ https://www.geometrictools.com/Documentation/LaplaceExpansionTheorem.pdf
+ */
+let matinv = (mat) =>
+  switch mat {
+  | [|m00, m01, m02, m10, m11, m12, m20, m21, m22|] =>
+    let det =
+      m00
+      *. m11
+      *. m22
+      +. m01
+      *. m12
+      *. m20
+      +. m02
+      *. m10
+      *. m21
+      -. m00
+      *. m12
+      *. m21
+      -. m01
+      *. m10
+      *. m22
+      -. m02
+      *. m11
+      *. m20;
+    assert (det != 0.);
```

Comment:
Is this assert redundant with the check in the following switch?
we can probably just do 
```
if (det == 0.) {
  invalid_arg("The current transform matrix cannot be inverted")
};
... do contents of switch
```
and remove the second switch?

Reply:
Oh oops, I had the assert before adding the switch and forgot to delete the assert. Will delete.

It could also be a simple if/then, but these days refmt want to make those ternary conditionals which I find always throws me off. Can do whichever though.

Reply:
Just thought that since the switch isn't actually destructuring anything, just checking for 0, that you could check for the error and then continue without needing to nest anything.  Not a big deal, will merge when assert is removed.

Reply:
Oh right! I forgot about this form of `if` and must have misread your suggestion at first. I like yours. Updated in c0dff54.